components/log-viewer-webui/client/src/pages/IngestPage/Jobs/utils.ts

```diff
@@ -0,0 +1,118 @@
+import dayjs from "dayjs";
+
+import {JobData} from "../Jobs/typings";
+import {QueryJobsItem} from "./sql";
+
+
+/**
+ * Enum of compression job statuses, matching the `CompressionJobStatus` class in
+ * `job_orchestration.scheduler.constants`.
+ *
+ * @enum {CompressionJobStatus}
+ */
+enum CompressionJobStatus {
+    PENDING = 0,
+    RUNNING = 1,
+    SUCCEEDED = 2,
+    FAILED = 3,
+}
+
+/**
+ * Convert JobStatus to Antd's badge string.
+ *
+ * @param status
+ * @return
+ * @throws an Error if status is not a CompressionJobStatus.
+ */
+const convertJobStatusToBadgeString = (status: CompressionJobStatus) => {
```

Comment:
can this be achieved by a map. e.g.,
const JOB_STATUS_TO_DISPLAY_NAME: Record<CompressionJobStatus, string> = Object.freeze({
[CompressionJobStatus.PENDING]: "warning",
...
})